File: Creational/FactoryMethod/TypeScript/Structure/ConcreteCreatorB.ts
import { Creator } from "./Creator";
import { ConcreteProductB } from "./ConcreteProductB";

/**
 * ConcreteCreatorB - Concrete implementation of Creator
 * Creates ConcreteProductB instances
 */
export class ConcreteCreatorB extends Creator {
  factoryMethod(): ConcreteProductB {
    return new ConcreteProductB();
  }
}

File: Creational/FactoryMethod/Java/Structure/ConcreteCreatorB.java
/**
 * ConcreteCreatorB - implementation cụ thể khác của Creator
 * Override factory method để tạo ConcreteProductB
 */
public class ConcreteCreatorB extends Creator {
    
    @Override
    public Product factoryMethod() {
        System.out.println("ConcreteCreatorB: Tạo Product B");
        return new ConcreteProductB();
    }
}
